test_crate.rs

```diff
@@ -0,0 +1,6 @@
+use std::marker::PhantomData;
+
+pub struct Foo<A, B>(PhantomData<(A, B)>);
+
+pub fn bleh<A=i32, X=char>() -> Foo<A, X> { Foo(PhantomData) }
+
```

Comment:
Yeah I added a commit to move things to my build server, rebasing the final error message work now.